### ISSUE TITLE
`RunTestExecutable` should not extend `AbstractExecTask`

### ISSUE THREAD

**Issue by lacasseio** [gradle/gradle-native#1010]
### Expected Behavior

Although the `RunTestExecutable` is executing the executable, how it is accomplished is an implementation detail. There are lots of methods that don't make sense in the context of test execution, for example, `ignoreExitValue` vs `ignoreFailures`. The later makes sense, but `ignoreExitValue` doesn't.

### Current Behavior

`RunTestExecutable` extends `AbstractExecTask`.

### Context

This extension was added when Google Test support was introduced in the software model in order to configure the test executable arguments and environment variable. Back then, we should have just added the appropriate fields.

### Your Environment

 * Build scan URL:

**Comment by big-guy** [comment]
I think this is fine to do, but even better would be to use a different task that's more `Test`-like like `XcTest`.

**Comment by lacasseio** [comment]
I agree, we should look at aligning Java test task, Xctest and RunTestExecutable together.
